components/host-dashboard/reports-section/TransactionsOverviewSection.js

```diff
@@ -270,9 +224,28 @@ const TransactionsOverviewSection = ({ host, isLoading, dateInterval }) => {
     },
   ];
 
+  /*
+   * If a date doesn't have any contributions or expenses API returns nothing.
+   * But we need to make sure the two series (expenses and contributions) show 0 in these cases rather than NaN which
+   * is shown by default by Appex charts.
+   */
+  if (series[0]?.data && series[1]?.data) {
+    series[0].data.forEach(data => {
+      if (!series[1].data.find(point => point.x === data.x)) {
+        series[1].data.push({ x: data.x, y: 0 });
+      }
+    });
+    series[1].data.sort((a, b) => new Date(a.x) - new Date(b.x));
+    series[1].data.forEach(data => {
+      if (!series[0].data.find(point => point.x === data.x)) {
+        series[0].data.push({ x: data.x, y: 0 });
+      }
+    });
+    series[0].data.sort((a, b) => new Date(a.x) - new Date(b.x));
+  }
```

Comment:
:ice_cream: **Not a requested change**: this is a bit difficult to read. `difference` from lodash could have helped to write something more explicit:

```es6
  const keysMissingFromSeries0 = differenceBy(series[1].data, series[0].data, 'x');
  const keysMissingFromSeries1 = differenceBy(series[0].data, series[1].data, 'x');
  series[0].push(...keysMissingFromSeries0.map(x => ({ x, y: 0 })));
  series[1].push(...keysMissingFromSeries1.map(x => ({ x, y: 0 })));
  series[0].data.sort((a, b) => new Date(a.x) - new Date(b.x));
  series[1].data.sort((a, b) => new Date(a.x) - new Date(b.x));
```